components/Auth/AuthModal.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+} from '@chakra-ui/react'
+import React, { useEffect } from 'react'
+import { AuthContainer, useUser } from '.'
+
+interface Props {
+  isOpen: boolean
+  onClose: () => void
+  message: string
+}
+
+const AuthModal = ({ isOpen, onClose, message }: Props): JSX.Element => {
+  const { userData } = useUser()
+
+  useEffect(() => {
+    if (userData) {
+      onClose()
+    }
+  }, [onClose, userData])
```

Comment:
What's the point of this effect? Seems like it automatically closes the modal if `userData` is defined?

Reply:
It doesn't close when you use the `GoogleSignInComponent` to authenticate yourself. At least that's what happens on my machine.

Reply:
Hmm, I haven't ran this locally but is that expected? I feel like if you use the `GoogleSignInComponent` to sign-in, you'd want this modal to close?

Also, I still think have the modal auto-close in generally is good UX - couldn't we just display a sign-out option if they were signed in?

Reply:
Yeah it makes sense for it to stay open even after you log in based on how Chakras `Modal` works. You sign in through a new broswer using the `GoogleSignInComponent` so once when that browser closes and you are authenticated the modal needs to know that it should close.  So basically once we have some sort of userData we can close the modal. Lmk if this makes sense.

Reply:
Oh, I see - so the `GoogleSignInComponent` also updates `userData`? If both login methods are getting us `userData` and when we wanna close the modal when we have that I think that makes sense.